operators/tokenizer/bpe_utils.hpp

```diff
@@ -230,3 +230,70 @@ class TokenWithRegularExp {
  private:
   std::u32string_view m_text;
 };
+
+class LRUCache {
+  // Store keys of cache
+  std::list<std::string> dq;
+
+  // Store references of keys in cache for efficiency
+  std::unordered_map<std::string, std::list<std::string>::iterator> references;
```

Comment:
why is the map unordered?

Reply:
because that one is just for the iterators for the keys (address of keys). `dq` is what we are using as a queue for the Queue+Map LRU implementation.